scpdt/plugin.py

```diff
@@ -256,12 +258,13 @@ def run(self, test, compileflags=None, out=None, clear_globs=False):
             *unless* the `mpl()` context mgr has a chance to filter them out
             *before* they become errors in `config.user_context_mgr()`.
             """
+            dt_config = config.dt_config
+
             with np_errstate():
-                with config.dt_config.user_context_mgr(test):
+                with dt_config.user_context_mgr(test):
                     with matplotlib_make_nongui():
-                        # XXX: might want to add the filter to `testmod`, too
-                        with warnings.catch_warnings():
-                            warnings.filterwarnings("ignore", category=DeprecationWarning)
+                        # XXX: local_resourses needed? they seem to be, w/o pytest
+                        with temp_cwd(test, dt_config.local_resources):
```

Comment:
@Sheila-nk also this: to you happen to remember why you preferred a global state + `copy_local_resources` at the  `DTmodule.collect` stage instead of the context manager?

Reply:
I had to go back to this PR: https://github.com/ev-br/scpdt/pull/85 to confirm.
https://github.com/ev-br/scpdt/pull/85#issuecomment-1672941580: The global variable `copied_files` allowed us to perform a proper cleanup of copied files after testing is complete. 
I am unsure why we needed the `copy_local_resources` function at the time, instead of the context manager. I will look into this a bit more.

Reply:
That would be great Sheila! 
IIUC, the need for a global list and a deducated function can be gone if tests are run in a temp dir. 
Something is fishy with the paths in the local_resources, see my comment below.